### PR TITLE
Fixes and improvements to chunk on topic groups

### PR DESCRIPTION
**Pull request by robander**
This fixes several problems with `@chunk` attributes that are specified on topic groups (`<topicgroup>` but also any `topicref` that specifies `@chunk` but does not specify a topic).

- When `@chunk` is specified on a topic group, `ChunkMapReader` will create a stub topic (or "stump" topic) and use that as the basis for any combined chunk.
- If `chunk="to-content"` is specified on any reference, processing is passed to  `ChunkTopicParser`, which reads child topic references and merges them.
- `ChunkTopicParser` assumes that any nested reference that specifies `@chunk` will also refer to a topic. If `@chunk` is specified on a group, it results in an NPE, as reported in #2730 and #2843.
- The first update here fixes the NPE. Before processing is passed over to `ChunkTopicParser`, it checks child topicrefs for the group condition + `@chunk`. If a group is chunked inside of the new chunk, this update will generate the stub ahead of time, avoiding the NPE entirely.
- The second trivial update -- I noticed that stub topics were generated that set the DITAArchVersion to 1.2, but this should be 1.3 at this point.
- The final enhancement is for topicgroups that need stub topics, but do not have a title. Today, for that condition, the name of the stub topic (such as `Chunk1234567`) becomes the actual title of the topic, so we generate output that includes the generated file name as a title. The update checks for stubs generated from a topic group with no title; if there is no title and no shortdesc, the stub is set only to `<dita></dita>`, which is all that we need for merging child content but does not have the side effect of bad titles.
- For the edge case condition of a topic group with no title (but with a shortdesc), this update generates the topic (with shortdesc) but leaves the title empty.

Tested the full build with the sample files in #2843; updated unit tests to include those same samples (as part of the existing topicgroup test); updated all test files to account for the version change and the group title change.

### CODE REVIEW

**Comment by drmacro** [comment]
Shouldn't the shortdesc for a topicgroup be ignored just as its title is? It couldn't possibly make sense to generate an empty-title topic in that case.

If the spec doesn't explicitly handle this case then that's an oversight.

File: src/main/java/org/dita/dost/reader/ChunkMapReader.java [hunk 1]
```diff
@@ -535,9 +541,24 @@ private void processCombineChunk(final Element topicref) {
         final ChunkTopicParser chunkParser = new ChunkTopicParser();
         chunkParser.setLogger(logger);
         chunkParser.setJob(job);
+        createChildTopicrefStubs(getChildElements(topicref, MAP_TOPICREF));
         chunkParser.setup(changeTable, conflictTable, topicref, chunkFilenameGenerator);
         chunkParser.write(currentFile);
     }
+    
+    //Before combining topics in a branch, ensure any descendant topicref with @chunk and no @href has a stub
```
Comment (jelovirt) [inline src/main/java/org/dita/dost/reader/ChunkMapReader.java]:
```suggestion
    /** Before combining topics in a branch, ensure any descendant topicref with @chunk and no @href has a stub */
```

**Comment by robander** [comment]
@drmacro probably, if actually using `<topicgroup>`, though as you noted it's something the spec forgot to address. I'd consider that a lower-priority ugly edge case of "why are you marking up your documents this way", and don't feel a strong need to add exceptions for it. 

Today, if you have `<topicgroup>` with a short description inside of a branch that's already chunked, you get the NPE; if you create a new chunk with the same markup, you get both a generated title like "Chunk123456" *and* the shortdesc, which I think is worse than just the shortdesc.

In general though, I was using "topic group" in the more general sense of "grouping element that has no href / no navtitle / but has shortdesc". The test case for this that went into the unit test was basically the following; definitely still an edge case, but one where I think the short description should not be dropped:
```
<topicref chunk="to-content">
  <topicmeta>
    <shortdesc>This is an edge case, but shouldn't crash the build</shortdesc>
  </topicmeta>
  <topicref href="shortdesc-kid.dita"/>
</topicref>
```

**Comment by robander** [comment]
Looking at other open bugs tagged with `preprocess/chunking`, I think this may also resolve #2787 and #2428  (@raducoravu you'd already suggested 2428 might be a duplicate of an earlier one, I'm still not sure if it was but I do think the larger fix here resolves any remaining issues there).
